### PR TITLE
Enable restoring Teleporter archives from the command line

### PR DESCRIPTION
**Pull request by iain-henderson**
…d line

## Thank you for your contribution to the Pi-hole Community!

Please read the comments below to help us consider your Pull Request.

We are all volunteers and completing the process outlined will help us review your commits quicker.

**Please make sure you**

 1. Base your code and PRs against the repositories developmental branch.
 2. [Sign Off](https://docs.pi-hole.net/guides/github/how-to-signoff/) all commits as we enforce the [DCO](https://docs.pi-hole.net/guides/github/dco/) for all contributions
 3. [Sign](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits) all your commits as they must have verified signatures
 4. File a pull request for any change that requires changes to [our documentation](https://docs.pi-hole.net/) at our [documentation repo](https://github.com/pi-hole/docs) 

---

**What does this PR aim to accomplish?:**

In combination with [updates to AdminLTE](https://github.com/pi-hole/AdminLTE/pull/2449) this PR enhances pihole so that it can restore a teleporter archive from the command line.

**How does this PR accomplish the above?:**

The addition of an argument to the teleporter arguments (i.e. `pihole -a -t -r myname.tar.gz`). This addition includes some additional argument parsing and help text.

**Link documentation PRs if any are needed to support this PR:**

https://github.com/pi-hole/docs/pull/818

---
**By submitting this pull request, I confirm the following:**

1. I have read and understood the [contributors guide](https://docs.pi-hole.net/guides/github/contributing/), as well as this entire template. I understand which branch to base my commits and Pull Requests against.
2. I have commented my proposed changes within the code and I have tested my changes.
3. I am willing to help maintain this change if there are issues with it later.
4. It is compatible with the [EUPL 1.2 license](https://opensource.org/licenses/EUPL-1.1)
5. I have squashed any insignificant commits. ([`git rebase`](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html))
6. I have checked that another pull request for this purpose does not exist.
7. I have considered, and confirmed that this submission will be valuable to others.
8. I accept that this submission may not be used, and the pull request closed at the will of the maintainer.
9. I give this submission freely, and claim no ownership to its content.

---
- [X] I have read the above and my PR is ready for review. *Check this box to confirm*

### CODE REVIEW

File: advanced/Scripts/webpage.sh [hunk 1]
```diff
@@ -617,6 +639,15 @@ Teleporter() {
     php "${webroot}/admin/scripts/pi-hole/php/teleporter.php" > "${filename}"
 }
 
+Restore() {
+    if [[ "${#args[@]}" -gt 3 ]]; then
+        # webroot is sourced from basic-install above
+        php "${webroot}/admin/scripts/pi-hole/php/teleporter.php" ${args[@]:3}
```
Comment (rdwebdesign) [inline advanced/Scripts/webpage.sh:645]: Stickler is complaining:

```suggestion
        php "${webroot}/admin/scripts/pi-hole/php/teleporter.php" "${args[@]:3}"
```

**Review by rdwebdesign — DISMISSED** [review]

**Comment by dschaper** [comment]
@rdwebdesign In case you missed this.

**Review by rdwebdesign — DISMISSED** [review]

File: advanced/Scripts/webpage.sh [hunk 2]
```diff
@@ -603,6 +604,27 @@ Interfaces:
 }
 
 Teleporter() {
+    case "${args[2]}" in
+        "-h" | "help"         ) TeleporterHelpFunc;;
+        "-r" | "restore"      ) Restore;;
```
Comment (yubiuser) [inline advanced/Scripts/webpage.sh:609]:
```suggestion
        "-r" | "restore"      ) TeleporterRestore;;
```

File: advanced/Scripts/webpage.sh [hunk 3]
```diff
@@ -617,6 +639,15 @@ Teleporter() {
     php "${webroot}/admin/scripts/pi-hole/php/teleporter.php" > "${filename}"
 }
 
+Restore() {
```
Comment (yubiuser) [inline advanced/Scripts/webpage.sh:642]:
```suggestion
TeleporterRestore() {
```

File: advanced/Scripts/webpage.sh [hunk 4]
```diff
@@ -603,6 +604,27 @@ Interfaces:
 }
 
 Teleporter() {
+    case "${args[2]}" in
+        "-h" | "help"         ) TeleporterHelpFunc;;
+        "-r" | "restore"      ) Restore;;
+        *                     ) Backup;;
```
Comment (yubiuser) [inline advanced/Scripts/webpage.sh:610]:
```suggestion
        *                     ) TeleporterBackup;;
```

File: advanced/Scripts/webpage.sh [hunk 5]
```diff
@@ -603,6 +604,27 @@ Interfaces:
...
+
+Example: pihole -a -t myname.tar.gz
+Set options for Teleporter
+
+Options:
+  -r, restore myname.tar.gz       Restore configuration from an archive with name myname.tar.gz as specified"
+    exit 0
+}
+
+Backup() {
```
Comment (yubiuser) [inline advanced/Scripts/webpage.sh:627]:
```suggestion
TeleporterBackup() {
```

**Review by yubiuser — CHANGES_REQUESTED** [review]
The change in `basic-install.sh` does not belong in your PR. Please rebase on latest `development`ans squash all your commits down.

**Comment by hubertbanas** [comment]
Thank you for this PR! I see potential for multi Pi-hole sync implementation for high availability.

When can we expect this PR to be merged?

**Comment by iain-henderson** [comment]
My understanding is that these changes mirror an implementation in the next major release of Pi-Hole (but that could be a misunderstanding).

**Pull request closed without merging**